packages/calypso-e2e/src/lib/pages/me/account-settings-page.ts

```diff
@@ -31,13 +32,19 @@ export class AccountSettingsPage {
 	 * Closes the currently logged in user's account.
 	 */
 	async closeAccount(): Promise< void > {
-		// This async navigation can mess with clicking the next button, so we need to make sure to wait explicitly for that async nav to commplete.
+		// Wait for the async navigation
 		await Promise.all( [
 			this.page.waitForNavigation(),
 			this.page.click( selectors.closeAccountLink ),
 		] );
 
+		// This page is tricky. All the text and the close account button load in and are "visible",
+		// except they are then covered over with gray boxes. The button is still "clickable", but doesn't do anything.
+		// The only thing that doesn't appear until all the loading is done is the sidebar of items to be deleted.
+		// So we must wait for that text before continuing, or our close account button click can get swallowed!
```

Comment:
Interesting find. I assume you've already tried other approaches and this was the best approach?
- waiting for `networkidle`
- wait for `closeAccountButton` with state {'enabled'}
- waiting for last request/response

There's also a method in `element-helper` that is `waitUntilElementEnabled` or something similar. I am guessing the element here isn't 'disabled' per se?

Reply:
Yeah, I personally think this is the best approach for right now!

I think it's more stable (a text selector) than waiting on a specific network response or networkidle, which i've found to be slightly more fussy recently. 

And yeah, this button is technically enabled 😱 . TBH, I don't think this is _quite_ implemented correctly in the front end - a button shouldn't be "disabled" but not actually marked as such in the DOM. For now, it's literally just an added gray box and handlers that do nothing 😦

Reply:
As discussed - this is a pseudo-element so we can't wait for the grey box to disappear, nor we can wait for the button to be enabled (because it already technically is).

This is the next best alternative.